grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1079,8 +1079,10 @@ class JsonController {
                     type : product.class,
                     url  : request.contextPath + "/" + type + "/redirect/" + product.id,
                     value: product.name,
-                    label: product.productCode + " " + product.name + " " + quantity,
-                    color: product.color
+                    label: product.productCode + " " + (product.translatedName ?: product.name) + " " + quantity,
+                    color: product.color,
+                    // Do not remove double negation even if IDE recommends this - we want to convert string/null to boolean
+                    hasTranslatedName: !!product.translatedName
```

Comment:
Why you cant return the translated name and then check if it is there if you need it? This one looks like a redundant field.

Reply:
because the one we are actually using to display the name is `label`, and it would be impossible to determine on the frontend side if inside this `label` we have `translatedName` or the original name

Reply:
ok, makes sense